components/AuthorizedPage/AuthorizedPage.tsx

```diff
@@ -0,0 +1,57 @@
+import { ReactNode, useEffect, useMemo, useState } from "react";
+import useApp from "../../hooks/useApp";
+import jwt from "jsonwebtoken";
+import { isInIframe } from "../../utils/misc";
+import { DashboardTokenPayload } from "../../lib/middlewares";
+import AccessWarning from "../AccessWarning/AccessWarning";
+import LoadingPage from "../LoadingPage/LoadingPage";
+
+type AuthorizedPageProps = {
+  children: ReactNode;
+};
+
+const AuthorizedPage = ({ children }: AuthorizedPageProps) => {
+  const app = useApp()?.getState();
+
+  // Next and React 18 have issues with hydatation
+  // GH Issue: https://github.com/vercel/next.js/discussions/35773
+  // To be fixed once a proper solution is available
+  const [mounted, setMounted] = useState(false);
+  useEffect(() => setMounted(true), []);
+
+  const tokenClaims = useMemo(() => {
```

Comment:
that's so generic, that should at some point also end up in `app-bridge`

Reply:
I remember I wanted to do that months ago but I think Kuba M. at the time wanted it outside of app-bridge

Reply:
@magul @Cloud11PL @krzysztofwolski 

Maybe we should have similar discussion about app-bridge scope, like we will have for app-sdk. Can you maybe track it somewhere, maybe create issue in app-bridge?

In general I'd like discussions like this to be moved out of PR (if we don't agree) but also, track it and decide eventually